reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/MismatchedLaterality.java

```diff
@@ -3,199 +3,126 @@
+        }
+    }
+
+    public void runJob() throws TermServerScriptException {
+        if (jobRun.getParamBoolean("notYetMembers")) {
+            // Process lateralisable Concepts not yet added to 723264001 |Lateralisable body structure reference set|.
+            String byLaterality = "( (<< 91723000 |Anatomical structure (body structure)| : 272741003 | Laterality (attribute) | = 182353008 |Side (qualifier value)|) MINUS ( * : 272741003 | Laterality (attribute) | = (7771000 |Left (qualifier value)| OR 24028007 |Right (qualifier value)| OR 51440002 |Right and left (qualifier value)|) ) )  MINUS (^ 723264001)";
+            String byHierarchy = "(( << 91723000 |Anatomical structure (body structure)| MINUS (* : 272741003 | Laterality (attribute) | = (7771000 |Left (qualifier value)| OR 24028007 |Right (qualifier value)| OR 51440002 |Right and left (qualifier value)|)))  AND (<  (^ 723264001)))   MINUS (^ 723264001)";
+            Set<Concept> notYetMembers = getConceptsByECL(byLaterality, byHierarchy);
+
+            reportOddNumberOfLateralisedChildren(notYetMembers, false);
+        }
+
+        if (jobRun.getParamBoolean("alreadyMembers")) {
+            // Process lateralisable Concepts already added to 723264001 |Lateralisable body structure reference set|.
+            String byMembership = "^ 723264001";
+            Set<Concept> alreadyMembers = getConceptsByECL(byMembership);
+
+            reportOddNumberOfLateralisedChildren(alreadyMembers, true);
+        }
+    }
+
+    private Set<Concept> getConceptsByECL(String... eclStatements) throws TermServerScriptException {
+        Set<Concept> concepts = new HashSet<>();
+        for (String eclStatement : eclStatements) {
+            if (jobRun.getParamBoolean("legacy")) {
+                concepts.addAll(findConcepts(eclStatement));
+            } else {
+                concepts.addAll(findConceptsWithoutEffectiveTime(eclStatement));
+            }
+        }
+
+        return concepts;
+    }
+
+    private void reportOddNumberOfLateralisedChildren(Set<Concept> lateralisableConcepts, boolean memberOfLateralisableReferenceSet) throws TermServerScriptException {
+        int counter = 0;
+        int size = lateralisableConcepts.size();
+        String isMember = memberOfLateralisableReferenceSet ? "Y" : "N";
+        info(String.format("%d lateralisable Concepts will be checked they do not have an odd number of children.", size));
+        for (Concept lateralisableConcept : lateralisableConcepts) {
+            counter++;
+            info(String.format("Processing %d/%d lateralisable Concepts with membership of 723264001 being %s.", counter, size, isMember));
+            if (!lateralisableConcept.isActive()) {
+                report(PRIMARY_REPORT, lateralisableConcept.getConceptId(), lateralisableConcept.getFsn(), lateralisableConcept.getSemTag(), isMember, "Required", "Concept is inactive and should be removed from reference set.");
+                continue;
+            }
```

Comment:
This situation may occur if the user runs the report before running the 'update lat' script.